src/main/java/database/models/FormplayerCaseIndexTable.java

```diff
@@ -0,0 +1,371 @@
+        String[] args = new String[]{indexName, targetValue};
+        Connection connection = null;
+        PreparedStatement selectStatement = null;
+        try {
+            connection = dataSource.getConnection();
+            selectStatement = SqlHelper.prepareTableSelectStatement(connection,
+                    TABLE_NAME,
+                    new String[]{COL_INDEX_NAME, COL_INDEX_TARGET},
+                    args);
+            ResultSet resultSet = selectStatement.executeQuery();
+            LinkedHashSet<Integer> ret = new LinkedHashSet<>();
+            SqliteIndexedStorageUtility.fillIdWindow(resultSet, COL_CASE_RECORD_ID, ret);
+            return ret;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (selectStatement != null) {
+                try {
+                    selectStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * Get a list of Case Record id's for cases which index any of a set of provided values
+     *
+     * @param indexName      The name of the index
+     * @param targetValueSet The set of cases targeted by the index
+     * @return An integer array of indexed case record ids
+     */
+    public LinkedHashSet<Integer> getCasesMatchingValueSet(String indexName, String[] targetValueSet) {
+        String[] args = new String[1 + targetValueSet.length];
+        args[0] = indexName;
+
+        System.arraycopy(targetValueSet, 0, args, 1, targetValueSet.length);
+        String inSet = getArgumentBasedVariableSet(targetValueSet.length);
+
+        String whereExpr = String.format("%s = ? AND %s IN %s", COL_INDEX_NAME, COL_INDEX_TARGET, inSet);
+
+        Connection connection = null;
+        PreparedStatement selectStatement = null;
+        try {
+            connection = dataSource.getConnection();
+            selectStatement = SqlHelper.prepareTableSelectStatement(connection,
+                    TABLE_NAME,
+                    whereExpr,
+                    args);
+            ResultSet resultSet = selectStatement.executeQuery();
+            LinkedHashSet<Integer> ret = new LinkedHashSet<>();
+            SqliteIndexedStorageUtility.fillIdWindow(resultSet, COL_CASE_RECORD_ID, ret);
+            return ret;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (selectStatement != null) {
+                try {
+                    selectStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
```

Comment:
print, going to stop commenting on these, but would be nice to handle these errors